lib/elixir/lib/kernel/typespec.ex

```diff
@@ -947,8 +947,11 @@ defmodule Kernel.Typespec do
 
       [head | tail] ->
         case Macro.expand_once(head, env) do
```

Comment:
I think then we need to replace this by expand instead of expand_once. Otherwise if you have a macro this macro:

```
defmacro current_module do
  quote do
    __MODULE__
  end
end
```

and this typespec:

```
current_module().Foo.Bar
```

It will no longer work, as the first expansion will return `__MODULE__`.

Reply:
> I think then we need to replace this by expand instead of expand_once.

Done in [3abb13b](https://github.com/elixir-lang/elixir/pull/14155/commits/3abb13b6d06cf36f73f63258bf3ebd324e4d8b05)

> It will no longer work, 

Just FYI, I checked and this was an infinite loop as well so technically it did never worked (we were returning the pre-expansion alias as is and kept looping). I agree we should fix it of course.